core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -2714,6 +2714,21 @@ public static long getSizeOfSegment(String tablePath, Segment segment) throws IO
     return size;
   }
 
+  public static long getIndexSize(String tablePath, Segment segment) throws IOException {
+    if (segment.getSegmentFileName() != null) {
+      SegmentFileStore fileStore = new SegmentFileStore(tablePath, segment.getSegmentFileName());
+      if (fileStore.getLocationMap() != null) {
+        fileStore.readIndexFiles(FileFactory.getConfiguration());
+        return getCarbonIndexSize(fileStore, fileStore.getLocationMap());
+      }
+    } else {
+      HashMap<String, Long> dataSizeAndIndexSize = getDataSizeAndIndexSize(
```

Comment:
still getting both data and index size !!
can get only index size here.

Reply:
can get the index file name and pass here ? again doing list files on S3a can be costly operation

Reply:
this case is not required, as this flow is for updating index for partition table. Removed this case